### ISSUE TITLE
Do not delete comments when removing unused declarations

### ISSUE THREAD

**Issue by dialex** [microsoft/TypeScript#34568]
- VSCode Version: 1.39.2
- OS Version: macOS 10.14.6

![demo](https://user-images.githubusercontent.com/2140450/67095635-50942380-f1ae-11e9-91b0-57963a28d44f.gif)

Steps to Reproduce:

1. Write a comment
2. Declare some JS variables and use them
3. Declare some unused JS variables
4. Click the tooltip and select "Delete al unused declarations

Expected:

- Comments are not deleted
- Unused declarations are removed

Actual

- Comments **are** deleted

Does this issue occur when all extensions are disabled?: Yes

**Comment by mjbvz** [comment]
This was previously reported in https://github.com/microsoft/TypeScript/issues/24758

Moving upstream for more feedback

**Comment by jessetrinity** [comment]
@mjbvz Which comments are desirable to remove? There seems to be some opinion that only doc comments should be left in place, but the comment on #24758 indicates that all comments are desirable:

```ts
// section comment
const ununsed = 0;
// other comment
const used = 0;
```
becomes
```ts
// section comment
// other comment
const used = 0;
```

I assume variable declarations are not a special case and we want to do the same for methods?

```ts
// comment
function unused(){}
// comment
function used(){}
```
becomes
```ts
// comment
// comment
function used(){}
```

Imports are currently special cased so that comments preceding only the first import are preserved. Do we want to change this for imports as well?

```ts
// tslint:disable:no-console
import thing from "module"
// comment
import otherThing from "otherModule"
```

```ts
// tslint:disable:no-console
// comment
import used from "otherModule"
```

TL;DR where _shouldn't_ we leave leading comments when we delete an unused variable?
